src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java

```diff
@@ -379,19 +388,42 @@ private void addReferenceConfidenceAttributes(final VCWithNewAlleles vcPair, fin
                     annotationMap.put(key, values);
                 }
                 try {
-                    values.add(parseNumber(value.toString()));
+                    //TODO: Verify that the code paths that lead here guarantee that these are always INFO attributes ?
+                    values.add(parseNumericInfoAttributeValue(vcfInputHeader, key, value.toString()));
                 } catch (final NumberFormatException e) {
                     warning.warn(String.format("Detected invalid annotations: When trying to merge variant contexts at location %s:%d the annotation %s was not a numerical value and was ignored",vcPair.getVc().getContig(),vcPair.getVc().getStart(),p.toString()));
                 }
             }
         }
     }
 
-    private Comparable<?> parseNumber(String stringValue) {
-        if (stringValue.contains(".")) {
-            return Double.parseDouble(stringValue);
-        } else {
-            return Integer.parseInt(stringValue);
+    // Use the VCF header's declared type for the given attribute to ensure that all the values for that attribute
+    // across all the VCs being merged have the same boxed representation. Some VCs have a serialized value of "0"
+    // for FLOAT attributes, with no embedded decimal point, but we still need to box those into Doubles, or the
```

Comment:
Of all the terrible things that happen to annotations in htsjdk, at least it guarantees some decimal precision (VCFEncoder.formatVCFDouble()). I'm guessing this was an issue related to GenomicsDB output (which uses htslib)?  I mentioned this to Karthik a while ago (https://github.com/broadinstitute/gatk/pull/4047#issuecomment-356063579)  It should have been addressed by #4261, but I haven't tested that explicitly.  Not that I object to the belt and suspenders approach.

Reply:
@ldgauthier Good to know - thanks. At least two users have reported this, one of them seemed to [implicate bcftools](https://github.com/broadinstitute/gatk/issues/4633), but that may be a red herring. I do think htsjdk can produce such output though, since it appears to rely on the in-memory representation of the value provided by the caller, rather than the header, to determine the format. I’d love to tighten that up in the next round of htsjdk updates.